worlds/satisfactory/__init__.py

```diff
@@ -0,0 +1,268 @@
+
+        self.critical_path = CriticalPathCalculator(self.game_logic, self.critical_path_seed, self.options)
+        self.critical_path.calculate()
+
+        self.state_logic = StateLogic(self.player, self.options, self.critical_path)
+        self.items = Items(self.player, self.game_logic, self.random, self.options, self.critical_path)
+
+        starting_inventory: list[str] = self.options.starting_inventory_preset.get_selected_list()
+        for item_name in starting_inventory:
+            self.push_precollected_by_name(item_name)
+
+
+    def create_regions(self) -> None:
+        locations: list[LocationData] = \
+            Locations(self.game_logic, self.options, self.state_logic, self.items, self.critical_path).get_locations()
+        create_regions_and_return_locations(
+            self.multiworld, self.options, self.player, self.game_logic, self.state_logic, self.critical_path,
+            locations)
+
+
+    def create_items(self) -> None:
+        self.setup_events()
+
+        number_of_locations: int = len(self.multiworld.get_unfilled_locations(self.player))
+        precollected_items: list[Item] = self.multiworld.precollected_items[self.player]
+        
+        self.multiworld.itempool += \
+            self.items.build_item_pool(self.random, precollected_items, number_of_locations)
+
+
+    def set_rules(self) -> None:
+        resource_sink_goal: bool = "AWESOME Sink Points (total)" in self.options.goal_selection \
+                                or "AWESOME Sink Points (per minute)" in self.options.goal_selection
+
+        required_parts = set(self.game_logic.space_elevator_tiers[self.options.final_elevator_package.value - 1].keys())
+
+        if resource_sink_goal:
+            required_parts.union(self.game_logic.buildings["AWESOME Sink"].inputs)
+
+        self.multiworld.completion_condition[self.player] = \
+            lambda state: self.state_logic.can_produce_all(state, required_parts)
+
+
+    def collect(self, state: CollectionState, item: Item) -> bool:
+        change = super().collect(state, item)
+        if change and item.name in self.game_logic.indirect_recipes:
+            state.prog_items[self.player][self.game_logic.indirect_recipes[item.name]] = 1
+        return change
+
+
+    def remove(self, state: CollectionState, item: Item) -> bool:
+        change = super().remove(state, item)
+        if change and item.name in self.game_logic.indirect_recipes:
+            del state.prog_items[self.player][self.game_logic.indirect_recipes[item.name]]
+        return change
+
+
+    def fill_slot_data(self) -> dict[str, object]:
+        slot_hub_layout: list[list[dict[str, int]]] = []
+
+        for tier, milestones in enumerate(self.game_logic.hub_layout, 1):
+            slot_hub_layout.append([])
+            for milestone, parts in enumerate(milestones, 1):
+                 slot_hub_layout[tier - 1].append({})
+                 for part, amount in parts.items():
```

Comment:
```suggestion
                slot_hub_layout[tier - 1].append({})
                for part, amount in parts.items():
```
These are indented by one too many spaces